### PR TITLE
Extract more functionality into the public api

### PR DESCRIPTION
**Pull request by danmichaelo**
I split it into two methods since I have a use case myself I want to have symmetric mappings both ways without touching the hierarchical mappings.

  
I also made the inference of skos:related from skos:relatedMatch optional, per #53

### CODE REVIEW

File: skosify/infer.py [hunk 1]
```diff
@@ -24,7 +24,11 @@ def skos_topConcept(rdf):
 
 
 def skos_hierarchical(rdf, narrower=True):
-    """Infer skos:broader/skos:narrower (S25) but only keep skos:narrower on request."""
+    """Infer skos:broader/skos:narrower (S25) but only keep skos:narrower on request.
+
+    :param bool narrower: If set to False, skos:narrowMatch will not be added,
```
Comment (osma) [inline skosify/infer.py]: Shouldn't this be "skos:narrower will not be added"?

**Comment by osma** [comment]
Other than the comment on the docstring, looks good to me. @nichtich ?

File: skosify/infer.py (hunk 1 above)
Comment (danmichaelo) [reply skosify/infer.py]: Good catch! Fixed

**Comment by danmichaelo** [comment]
I found that I actually depended on `check_hierarchy` in one of my scripts, so I extracted functionality from that method into a new `check` namespace. Since the method was quite complex, I split it into three smaller methods at the same time. If you are not happy with the names of the methods, feel free to edit that!

I'm not sure if I like the re-calling of `setup_top_concepts(rdf, mark_top_concepts)` after `check.hierarchy_cycles()`, but I kept it for backwards compability. I guess it's there just in case the cycle breaking reveals new top concepts?

**Comment by danmichaelo** [comment]
The latest commit is not really related to the rest of the PR, sorry about that.

**Comment by osma** [comment]
> I'm not sure if I like the re-calling of setup_top_concepts(rdf, mark_top_concepts) after check.hierarchy_cycles(), but I kept it for backwards compability. I guess it's there just in case the cycle breaking reveals new top concepts?

That's exactly the reason. When I tested Skosify with a number of SKOS vocabularies back in 2011-2013 and wrote the EKAW and JoDS papers about it, there were some (can't remember which ones) that had a cycle on the top level. They were not detected as top concepts at first because all the concepts seemed to have a broader relationship.

Back then the Skosify processing was just a sequence of processing steps. Now that @nichtich and you have been pulling apart the processing steps and making them into a public API, the old assumptions are not valid anymore.

**Comment by danmichaelo** [comment]
@nichtich , do you have a chance to look at this?

**Comment by danmichaelo** [comment]
Rebased onto master

**Pull request merged**